docs/generate_docs.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# script for generating documentation for NMODL
+
+set -xeu
+
+# in order to create the docs, we first need to build NMODL in some dir
+build_dir="${1:-"$(mktemp -d)"}"
+if ! [ -d "${build_dir}" ]
+then
+    mkdir -p "${build_dir}"
+fi
+
+# the Python executable to use (default: whatever `python3` is)
+python_exe="${2:-"$(command -v python3)"}"
+
+# should we use a separate venv?
+use_venv="${3:-}"
+
+if [ "${use_venv}" != "false" ]
+then
+    venv_name="$(mktemp -d)"
+    $(command -v python3) -m venv "${venv_name}"
+    . "${venv_name}/bin/activate"
+    python_exe="$(command -v python)"
+    ${python_exe} -m pip install -U pip
+else
+    python_exe="$(command -v python3)"
+fi
+${python_exe} -m pip install ".[docs]" -C build-dir="$(cd "${build_dir}"; pwd -P)"
```

Comment:
How are the following different?
```
$(cd "${build_dir}"; pwd -P)
$(realpath "${build_dir}")
```

Reply:
`realpath` isn't available on some older versions of MacOS (see [here](https://apple.stackexchange.com/a/450116)), so the first one should work on both Linux and any version of MacOS (assuming the dir has the right permissions, which, for a temp one, should always be the case).